pkg/proxy/proxyinfo.go

```diff
@@ -54,22 +70,30 @@ func GetProxyInfo(kubeconfig, configContext, revision, istioNamespace string) (*
 			pi = append(pi, istioVersion.ProxyInfo{
 				ID:           ss.ProxyID,
 				IstioVersion: ss.SyncStatus.IstioVersion,
+				Type:         ConvertNodeType(ss.ProxyType),
 			})
 		}
 	}
 
 	return &pi, nil
 }
 
-// GetIDsFromProxyInfo is a helper function to retrieve list of IDs from Proxy.
+// GetIDsFromProxyInfo is a helper function to retrieve list of IDs from Proxy. Which contains only the data panel.
 func GetIDsFromProxyInfo(kubeconfig, configContext, revision, istioNamespace string) ([]string, error) {
 	var IDs []string
 	pi, err := GetProxyInfo(kubeconfig, configContext, revision, istioNamespace)
 	if err != nil {
 		return IDs, fmt.Errorf("failed to get proxy infos: %v", err)
 	}
 	for _, pi := range *pi {
+		if pi.Type == istioVersion.Ztunnel ||
+			pi.Type == istioVersion.Gateway ||
```

Comment:
What if the gateway is not installed by this iop, in this case, it is treated as `detached`